src/magic_folder/test/cli/common.py

```diff
@@ -26,6 +29,66 @@
     MagicFolderCommand,
     run_magic_folder_options,
 )
+from ...api_cli import (
+    MagicFolderApiCommand,
+    run_magic_folder_api_options,
+)
+
+
+
+@contextmanager
+def _pump_client(http_client):
+    """
+    Periodically flush all data in the HTTP client while this context manager
+    is active.
+
+    py:`treq.testing.RequestTraversalAgent` doesn't automatically pass data
+    between a client and server, if a response isn't generated synchronously,
+    so if we want requests to complete, we need to call ``.flush`` on it.
+
+    :param treq.client.HTTPClient http_client:
+        An HTTP Client wrapping a :py:`treq.testing.RequestTraversalAgent`.
+    """
+    def pump():
+        while True:
+            http_client._agent.flush()
+            yield
+
+    coop = Cooperator()
+    task = coop.cooperate(pump())
+    try:
+        yield
+    finally:
+        task.stop()
+        coop.stop()
+
+def _subclass_of(base):
+    def validator(inst, attr, value):
+        if not issubclass(value, base):
+            raise TypeError(
+                "'{name}' must be a subclass of {base!r}"
+                "(got {value!r}).".format(
+                    name=attr.name,
+                    base=base,
+                    value=value,
+                )
+            )
+
+    return validator
+
+
+@attr.s
+class Command(object):
+    """
+    A magic-folder command entry point.
+
+    :ivar Type[usage.Options] options: The options class to use.
+    :ivar Callable[[usage.Options], Deferred[None]] entry_point:
+        The function to pass the options to.
+    """
+    options = attr.ib(validator=_subclass_of(usage.Options))
```

Comment:
What's wrong with the built-in "instance_of"?

Reply:
This is not taking an object, but a class. `issubclass` is not `isinstance`, which is what these validators map to.